opentelemetry-sdk/src/metrics/exporter.rs

```diff
@@ -18,16 +18,16 @@ pub trait PushMetricsExporter: Send + Sync + 'static {
     /// implement any retry logic. All errors returned by this function are
     /// considered unrecoverable and will be reported to a configured error
     /// Handler.
-    async fn export(&self, metrics: &mut ResourceMetrics) -> Result<()>;
+    async fn export(&self, metrics: &mut ResourceMetrics) -> MetricResult<()>;
 
     /// Flushes any metric data held by an exporter.
-    async fn force_flush(&self) -> Result<()>;
+    async fn force_flush(&self) -> MetricResult<()>;
 
     /// Releases any held computational resources.
     ///
     /// After Shutdown is called, calls to Export will perform no operation and
     /// instead will return an error indicating the shutdown state.
-    fn shutdown(&self) -> Result<()>;
+    fn shutdown(&self) -> MetricResult<()>;
```

Comment:
Not related to this PR, but the export interface of metrics, logs and traces are not consistent with each other:
 - `SpanExporter::shutdown()` and `LogExporter::shutdown()` does't return the status of the shutdown. `MetricsExporter::shutdown` return status.
 - `LogsExporter::force_flush()` is missing. `SpanExporter::force_flush()` is not async method, while `MetricsExporter::force_flush()` is an async method.

We need to discuss and make them consistent separate of this PR.